### PR TITLE
[Promotion] Behats for start and ends date for promotion

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

File: features/promotion/receiving_discount/applying_promotion_with_expiration_date.feature [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    I want to have promotion's discounts applied to my cart only if the given promotion is valid
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store has a product "PHP T-Shirt" priced at "$100.00"
+        And there is a promotion "Christmas sale"
+        And this promotion gives "$10.00" discount to every order
+
+    @ui
+    Scenario: Receiving a discount from promotion which does not expire
```
Comment (TheMadeleine) [inline features/promotion/receiving_discount/applying_promotion_with_expiration_date.feature]: a promotion

File: features/promotion/receiving_discount/applying_promotion_with_expiration_date.feature [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        Given this promotion expires tomorrow
+        When I add product "PHP T-Shirt" to the cart
+        Then my cart total should be "$90.00"
+        And my discount should be "-$10.00"
+
+    @ui
+    Scenario: Receiving no discount from valid and expired promotion
```
Comment (TheMadeleine) [inline features/promotion/receiving_discount/applying_promotion_with_expiration_date.feature]: `a valid but exipred promotion`

File: features/promotion/receiving_discount/applying_promotion_with_expiration_date.feature [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        Given this promotion has already expired
+        When I add product "PHP T-Shirt" to the cart
+        And my cart total should be "$100.00"
+        And there should be no discount
+
+    @ui
+    Scenario: Receiving a discount from promotion which is already started
```
Comment (TheMadeleine) [inline features/promotion/receiving_discount/applying_promotion_with_expiration_date.feature]: `a promotion which has already started`

File: features/promotion/receiving_discount/applying_promotion_with_expiration_date.feature [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        Given this promotion has started yesterday
+        When I add product "PHP T-Shirt" to the cart
+        Then my cart total should be "$90.00"
+        And my discount should be "-$10.00"
+
+    @ui
+    Scenario: Receiving no discount from promotion that has not been started yes
```
Comment (TheMadeleine) [inline features/promotion/receiving_discount/applying_promotion_with_expiration_date.feature]: a promotion

Comment (TheMadeleine) [inline features/promotion/receiving_discount/applying_promotion_with_expiration_date.feature]: <del>yes</del> => **yet**

Comment (lchrusciel) [inline features/promotion/receiving_discount/applying_promotion_with_expiration_date.feature]: 🐋

**Comment by michalmarcinkowski** [comment]
👍

**Pull request merged**